app/experimenter/nimbus-ui/src/components/AppLayoutWithSidebar/index.test.tsx

```diff
@@ -108,5 +110,32 @@ describe("AppLayoutWithSidebar", () => {
       expect(overviewLink!).toHaveClass("text-dark");
       expect(overviewLink!).not.toHaveClass("text-primary");
     });
+
+    it("renders information about missing experiment details", async () => {
+      const review = {
+        ready: false,
+        invalidPages: ["overview", "branches", "metrics", "audience"],
+      };
+      renderWithRouter(
+        <MockedCache mocks={[mock]}>
+          <AppLayoutWithSidebar {...{ review }}>
+            <p data-testid="test-child">Hello, world!</p>
+          </AppLayoutWithSidebar>
+        </MockedCache>,
+      );
+
+      expect(screen.queryByTestId("missing-details")).toBeInTheDocument();
+
+      review.invalidPages.forEach((page) => {
+        page = page.toLowerCase();
+
+        expect(
+          screen.queryByTestId(`missing-detail-alert-${page}`),
+        ).toBeInTheDocument();
+        expect(
+          screen.queryByTestId(`missing-detail-link-${page}`),
+        ).toBeInTheDocument();
+      });
+    });
```

Comment:
Should we have a test for when `ready` is set to `true`? I know you added `expect(screen.queryByTestId("missing-details")).not.toBeInTheDocument();` to "renders expected URLs" but seems like we should have one to check if the "Review & Launch" link is disabled or enabled.

Reply:
I can do that (but of note, the link is not disabled or enabled, it is either rendered of not, but I can check that it's in the dom)